### PR TITLE
 Add the missing Serving Runtime pod sepc

### PR DESCRIPTION
**Pull request by tarukumar**
##### Short description:
Observed that some parameters in the Serving Runtime specification were missing.
##### More details:
Several parameters from the runtime manifest are not present in the runtime resource. (https://github.com/kserve/kserve/blob/master/pkg/apis/serving/v1alpha1/servingruntime_types.go)

1.  I added new parameters, spec_annotation and spec_labels, as they are specific to the model pod based on runtime  rather than the metadata. I used the prefix `spec` to enhance understanding.
 2.   Other parameters are standard ones that were missing from the codebase, as detailed in the manifest below.
##### What this PR does / why we need it:
This PR add the missing parameter for runtime CR
## Summary by CodeRabbit

- **New Features**
	- Enhanced configuration options for the ServingRuntime class, allowing users to specify additional parameters for Kubernetes pod scheduling and resource management.
	- New optional attributes include `spec_labels`, `spec_annotations`, `affinity`, `node_selector`, `tolerations`, and `volumes`.

- **Documentation**
	- Expanded documentation for the constructor to include descriptions of the newly added parameters.

### CODE REVIEW

File: ocp_resources/serving_runtime.py [hunk 1]
```diff
@@ -101,3 +120,21 @@ def to_dict(self) -> None:
...
+                _spec["labels"] = self.spec_labels
+
+            if self.spec_annotations:
+                _spec["annotations"] = self.spec_annotations
+
+            if self.affinity:
+                _spec["affinity"] = self.affinity
+
+            if self.node_selector:
+                _spec["nodeSelector"] = self.node_selector
```
[the remark this answers is not in the document]
Comment (tarukumar) [reply ocp_resources/serving_runtime.py:134]: This seems false positive

File: ocp_resources/serving_runtime.py
[the remark this answers is not in the document]
Comment (tarukumar) [reply ocp_resources/serving_runtime.py:1]: would you like me to close this pr

[the remark this answers is not in the document]
Comment (myakove) [reply ocp_resources/serving_runtime.py:1]: this is the output with the class-generator fixes

https://github.com/RedHatQE/openshift-python-wrapper/pull/2152/files#diff-b72dee67f8fd96073abc7b29686c8e53bab92357e56efb37ec9748402ab51a03

**Comment by myakove** [comment]
@tarukumar 
The file link is just a test, we still need to generate the resource after the merge.
I will do that once my PR is merged.

**Comment by tarukumar** [comment]
> @tarukumar The file link is just a test, we still need to generate the resource after the merge. I will do that once my PR is merged.
Doesn't it make sense to reopen this pr and merge it once you pr is merged

**Comment by tarukumar** [comment]
i will regenerate and send the pr here

**Comment by myakove** [comment]
> i will regenerate and send the pr here

My PR was merged

**Comment by tarukumar** [comment]
@myakove if you want you can send the PR i am out for couple of hour for important work

**Pull request closed without merging**